### PR TITLE
refactor(node): add generic message handler to adapter actors

### PR DESCRIPTION
**Pull request by tmpolaczyk**

### CODE REVIEW

File: node/src/signature_mngr.rs [hunk 1]
```diff
@@ -579,91 +580,19 @@ impl Actor for SignatureManagerAdapter {
...
-impl Handler<VerifySignatures> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<(), failure::Error>>;
+// Delegate all the SignatureManager messages to the inner SignatureManager
+impl<M> Handler<M> for SignatureManagerAdapter
+where
+    M: Message + Send + 'static,
+    <M as actix::Message>::Result: Send,
+    Result<<M as actix::Message>::Result, actix::MailboxError>:
+        FlattenResult<OutputResult = <M as actix::Message>::Result>,
+    SignatureManager: Handler<M>,
```
Comment (tmpolaczyk) [inline node/src/signature_mngr.rs:590]: If possible I would like this to be simplified as:

```suggestion
    DelegateHandler<M, SignatureManager>
```

Comment (tmpolaczyk) [reply node/src/signature_mngr.rs:590]: Not possible :disappointed:

**Review by lrubiorod — APPROVED** [review]

File: node/src/storage_mngr.rs [hunk 2]
```diff
@@ -11,11 +11,11 @@ use std::{
 use actix::prelude::*;
 use bincode::{deserialize, serialize};
 use futures::Future;
+use futures_util::FutureExt;
 
-use crate::config_mngr;
+use crate::{config_mngr, utils::FlattenResult};
```
Comment (lrubiorod) [inline node/src/storage_mngr.rs:16]: fmt

File: node/src/signature_mngr.rs [hunk 3]
```diff
@@ -12,6 +12,7 @@ use crate::{
     actors::storage_keys::{BN256_SECRET_KEY, MASTER_KEY},
     config_mngr, storage_mngr,
     utils::stop_system_if_panicking,
+    utils::FlattenResult,
```
Comment (lrubiorod) [inline node/src/signature_mngr.rs:15]: fmt

**Pull request merged**